src/pages/data/[projectId].tsx

```diff
@@ -0,0 +1,477 @@
+              />
+              <h1 className="text-left ml-3">Number of Replies</h1>
+            </div>
+            <div>
+              <div className="flex bg-gray-dark h-3">
+                <div className="bg-purple h-full w-10" />
+              </div>
+              <div className="h-[28rem] bg-white">
+                <TimeSeriesLineChart
+                  data={timeSeriesData?.length ? [timeSeriesData[0]] : []}
+                  yLabel="Number of comments"
+                  dateRange={dateRange}
+                />
+              </div>
+            </div>
+          </div>
+          <div id="user-engagement-replies" className=" w-full mb-8">
+            <div className="data-heading flex items-center mb-3">
+              <FontAwesomeIcon
+                className="text-yellow"
+                icon={faChartBar}
+                fixedWidth
+              />
+              <h1 className="text-left ml-3">User engagement - User replies</h1>
+            </div>
+            <div>
+              <div className="flex bg-gray-dark h-3">
+                <div className="bg-yellow h-full w-10" />
+              </div>
+              <div className="h-28 bg-white" />
+            </div>
+          </div>
+          <div id="user-engagement-likes" className=" w-full mb-8">
+            <div className="data-heading flex items-center mb-3">
+              <FontAwesomeIcon
+                className="text-green"
+                icon={faChartBar}
+                fixedWidth
+              />
+              <h1 className="text-left ml-3">User engagement - Likes</h1>
+            </div>
+            <div>
+              <div className="flex bg-gray-dark h-3">
+                <div className="bg-green h-full w-10" />
+              </div>
+              <div className="h-28 bg-white" />
+            </div>
+          </div>
+        </div>
+      </div>
+      <SquareField
+        squares={dataSquares.data.bottom}
+        className="hidden md:block z-20"
+      />
+    </div>
+  );
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const projects = await prisma.dataDashboardProject.findMany({
+    select: {
+      id: true,
+      label: true,
+    },
+  });
+  return {
+    paths: projects.map((project) => ({
+      params: {
+        projectId: project.id,
+      },
+    })),
+    fallback: false,
+  };
+};
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const projectId = context.params?.projectId as string;
+
+  // Prefetch
+  const projects = await prisma.dataDashboardProject.findMany({
+    select: {
+      id: true,
+      label: true,
+    },
+  });
+  await prisma.dataDashboardProject.findFirst({
+    where: {
+      id: projectId,
+    },
+    include: {
+      data: {
+        include: {
+          values: true,
+        },
+      },
+    },
+  });
+
+  return {
+    props: {
+      projects: projects ?? [],
+      projectId,
+    },
+    revalidate: 10,
+  };
+};
```

Comment:
Please use trpc in DataProject instead ( you can use the getAllRequests of the playground page as a reference )

Reply:
Do you want me to get rid of getStaticProps and getStaticPaths and fetch the data from just inside the component?

Reply:
Yeah that's what I meant. It's not bad practice or anything, just something we should keep consistent across our components/pages

Reply:
Done!